go/vt/vtcombo/tablet_map.go

```diff
@@ -293,7 +295,7 @@ func (itc *internalTabletConn) Execute(ctx context.Context, target *querypb.Targ
 	if err != nil {
 		return nil, tabletconn.ErrorFromGRPC(vterrors.ToGRPC(err))
 	}
-	return reply, nil
+	return reply.Copy(), nil
```

Comment:
This change plus the one in:
```
func (result *Result) Copy() *Result {
```
Does not seem related to this PR. Am I missing something here? If it's not related to this, can we move it to a different one ? 

It seems to me that commit `1f4d932e60e31b59dc4f859d48a11e7f2e9e438c` can be just a small PR.